lib/Sema/LookupVisibleDecls.cpp

```diff
@@ -478,6 +474,56 @@ static void
   lookupTypeMembers(BaseTy, PT, Consumer, CurrDC, LS, Reason);
 }
 
+/// Trigger synthesizing implicit member declarations to make them "visible".
+static void synthesizeMemberDeclsForLookup(NominalTypeDecl *NTD,
+                                           const DeclContext *DC) {
+  // Synthesize the memberwise initializer for structs or default initializer
+  // for classes.
+  if (!NTD->hasInterfaceType())
+    TypeChecker::addImplicitConstructors(NTD);
+
+  // Check all conformances to trigger the synthesized decl generation.
+  // e.g. init(rawValue:) for RawRepresentable.
+  auto resolver = NTD->getASTContext().getLazyResolver();
+  if (!resolver)
+    return;
+  for (auto Conformance : NTD->getAllConformances()) {
+    auto Proto = Conformance->getProtocol();
+    if (!Proto->isAccessibleFrom(DC))
+      continue;
+    auto NormalConformance = dyn_cast<NormalProtocolConformance>(
+        Conformance->getRootConformance());
+    if (!NormalConformance)
+      continue;
+    for (auto Member : Proto->getMembers()) {
+      if (auto *ATD = dyn_cast<AssociatedTypeDecl>(Member)) {
+        if (!NormalConformance->hasTypeWitness(ATD))
+          resolver->resolveTypeWitness(NormalConformance, ATD);
+      } else if (auto *VD = dyn_cast<ValueDecl>(Member)) {
+        if (!NormalConformance->hasWitness(VD))
+          resolver->resolveWitness(NormalConformance, VD);
+      }
+    }
+  }
+
+  // Generate '$' and '_' prefixed variables that have attached property
+  // wrappers.
+  auto synthesizePropertyWrappers = [](IterableDeclContext *IDC) {
+    for (auto Member : IDC->getMembers()) {
+      if (auto var = dyn_cast<VarDecl>(Member)) {
+        if (var->hasAttachedPropertyWrapper()) {
+          auto sourceFile = var->getDeclContext()->getParentSourceFile();
```

Comment:
You can check if the IDC itself is a member of a source file outside the loop.

Reply:
Thanks, done!